### PR TITLE
2648 Add LongestRescale transform

### PR DESCRIPTION
**Pull request by Nic-Ma**
part of #2648 .

### Description
This PR added the `LongestRescale` and `LongestRescaled` transforms, inspired by the albumentations API.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_inverse.py [hunk 1]
```diff
@@ -249,15 +250,6 @@
     )
 )
 
-TESTS.append(
```
Comment (Nic-Ma) [inline tests/test_inverse.py:252]: @rijobro I removed this test case here because it's duplicated with previous lines, seems like a copy typo.

Thanks.

**Comment by wyli** [comment]
I think there's a large overlap between this and `Resize`, `Resize` also supports `int` as the `spatial_size`. would be nice to merge this into `Resize`, and provide a flexible `size_mode` for `Resize`. maybe in the future with different `size_mode`, we'll have different interpretation for cases such as `spatial_size=32` and `spatial_size=(32, -1, -1)`

File: monai/transforms/spatial/array.py [hunk 2]
```diff
@@ -378,20 +390,25 @@ def __call__(
...
-        if output_ndim > input_ndim:
-            input_shape = ensure_tuple_size(img.shape, output_ndim + 1, 1)
-            img = img.reshape(input_shape)
-        elif output_ndim < input_ndim:
-            raise ValueError(
-                "len(spatial_size) must be greater or equal to img spatial dimensions, "
-                f"got spatial_size={output_ndim} img={input_ndim}."
-            )
-        spatial_size = fall_back_tuple(self.spatial_size, img.shape[1:])
+        if self.size_mode == "all":
```
Comment (wyli) [inline monai/transforms/spatial/array.py:393]: this could be enhanced with the `look_up_option` api...

**Review by wyli — APPROVED** [review]
thanks for the quick updates

File: monai/transforms/spatial/array.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/transforms/spatial/array.py:391]: Sure, updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @madil90 ,

May I know why the blossom CI failed? I tried to re-run it, but it still fails.

Thanks in advance.

**Pull request merged**